python/ray/tune/progress_reporter.py

```diff
@@ -1520,7 +1520,8 @@ def _detect_reporter(**kwargs) -> TuneReporterBase:
 
     Keyword arguments are passed on to the reporter class.
     """
-    if IS_NOTEBOOK:
+    _trainer_api = kwargs.pop("_trainer_api", False)
```

Comment:
We can add `_trainer_api` as a keyword argument separate from kwargs.

Reply:
Just added a keyword arg `_trainer_api` for the function `_detect_reporter()`.